### PR TITLE
fix(ClusterRules): Open rows memorization 

### PR DESCRIPTION
**Pull request by Fewwy**
Fixes https://issues.redhat.com/browse/CCXDEV-7900.

Recording the opened rows for the single cluster page
Fixed the test

The expand collapse behavior:
1)When the user opens the page for the first time - the first row is expanded
2)If he clicks sort after that, only the rows that were expanded will stay that way
if he expands several different rows\all rows before that -> only these rows will stay expanded

### CODE REVIEW

**Comment by gkarat** [comment]
There is no "enh" commit type. Refer to https://github.com/angular/angular/blob/main/CONTRIBUTING.md#type and https://github.com/semantic-release/semantic-release#commit-message-format.

File: src/Components/ClusterRules/ClusterRules.spec.ct.js [hunk 1]
```diff
@@ -186,9 +186,9 @@ describe('cluster rules table', () => {
   });
 
   describe('defaults', () => {
-    it('only first item expanded', () => {
```
Comment (gkarat) [inline src/Components/ClusterRules/ClusterRules.spec.ct.js:189]: This is expected that the first item is expanded by default, you can't edit the test like this.

File: src/Components/ClusterRules/ClusterRules.spec.ct.js [hunk 2]
```diff
@@ -207,6 +207,16 @@ describe('cluster rules table', () => {
     cy.get(EXPANDABLES).should('have.length', 0);
   });
 
+  it.only('expand one row then sort', () => {
```
Comment (gkarat) [inline src/Components/ClusterRules/ClusterRules.spec.ct.js:210]:
```suggestion
  it('expand one row then sort', () => {
```

File: src/Components/ClusterRules/ClusterRules.spec.ct.js [hunk 3]
```diff
@@ -207,6 +207,16 @@ describe('cluster rules table', () => {
...
+    cy.get(ROOT).find('#expandable-toggle0').click();
+    cy.get(ROOT)
+      .find('th[data-label=Description]')
+      .find('button')
+      .click()
+      .click();
+    cy.get('table').find('tr').find('td[id="expanded-content11"]');
```
Comment (gkarat) [inline src/Components/ClusterRules/ClusterRules.spec.ct.js:217]: The selector points to a component that is rendered *always* (try to play with a table where all rows are collapsed, the selector will still yield you a container) => this assertion doesn't do what you expect (will be always true).

File: src/Components/ClusterRules/ClusterRules.js [hunk 4]
```diff
@@ -125,7 +127,11 @@ const ClusterRules = ({ cluster }) => {
       .map((value, key) => [
         {
           rule: value,
-          isOpen: isAllExpanded,
+          isOpen: isAllExpanded
+            ? true
+            : value?.rule_id === recordExpanded.includes(value?.rule_id)
+            ? true
+            : false,
```
Comment (gkarat) [inline src/Components/ClusterRules/ClusterRules.js:134]: down with ternary overuse 
```suggestion
          isOpen: isAllExpanded || value?.rule_id === recordExpanded.includes(value?.rule_id);
```

**Review by gkarat — COMMENTED** [review]
This does fixes the bug, however breaks other things, see the video:

https://user-images.githubusercontent.com/31385370/166456462-b1ced222-e4bb-4e10-99fd-771bb6b6520e.mp4

File: src/Components/ClusterRules/ClusterRules.spec.ct.js (hunk 1 above)
Comment (Fewwy) [reply src/Components/ClusterRules/ClusterRules.spec.ct.js:189]: Its not expanded anymore, so why I shouldn't change the test

Comment (gkarat) [reply src/Components/ClusterRules/ClusterRules.spec.ct.js:189]: Showing the first recommendation expanded has been a business rule and a feature. There was no request to change it even though I see that RHEL Advisor stopped doing it for some reason. We need to keep the behavior as it has been.

File: src/Components/ClusterRules/ClusterRules.js [hunk 5]
```diff
@@ -119,13 +118,17 @@ const ClusterRules = ({ cluster }) => {
     setDisplayedRows(collapseRows);
   };
 
-  const buildFilteredRows = (allRows, filters) =>
-    allRows
+  const buildFilteredRows = (allRows, filters) => {
+    const expandedRows = displayedRows
+      .filter((ruleExpanded) => ruleExpanded?.isOpen === true)
+      .flatMap((object) => [object?.rule?.rule_id]);
```
Comment (gkarat) [inline src/Components/ClusterRules/ClusterRules.js:124]:
```suggestion
      .filter((ruleExpanded) => ruleExpanded.isOpen)
      .map((object) => object.rule.rule_id)
```

File: src/Components/ClusterRules/ClusterRules.js [hunk 6]
```diff
@@ -119,13 +118,17 @@ const ClusterRules = ({ cluster }) => {
...
+
+    return allRows
       .filter((rule) => passFilters(rule, filters))
       .map((value, key) => [
         {
           rule: value,
-          isOpen: isAllExpanded,
+          isOpen: isAllExpanded || expandedRows?.includes(value?.rule_id),
```
[the remark this answers is not in the document]
Comment (Fewwy) [reply src/Components/ClusterRules/ClusterRules.js]: Implemented

File: src/Components/ClusterRules/ClusterRules.js (hunk 5 above)
Comment (Fewwy) [reply src/Components/ClusterRules/ClusterRules.js]: Implemented

**Pull request merged**